Sources/AppcuesKit/Presentation/Extensions/UIBezierPath+Tooltip.swift

```diff
@@ -15,84 +15,118 @@ internal struct Pointer {
             let bottomRight = CGPoint(x: mainRect.maxX - cornerRadius, y: mainRect.maxY - cornerRadius)
-            addArc(withCenter: bottomRight, radius: cornerRadius, startAngle: 0, endAngle: .pi / 2, clockwise: true)
+            path.addArc(center: bottomRight, radius: cornerRadius, startAngle: 0, endAngle: .pi / 2, clockwise: false)
         }
 
         if case .bottom = pointer.edge {
-            addTriangle(triangle)
+            path.addTriangle(triangle)
         }
 
         if !triangle.overridesBottomLeftCorner {
             let bottomLeft = CGPoint(x: mainRect.minX + cornerRadius, y: mainRect.maxY - cornerRadius)
-            addArc(withCenter: bottomLeft, radius: cornerRadius, startAngle: .pi / 2, endAngle: .pi, clockwise: true)
+            path.addArc(center: bottomLeft, radius: cornerRadius, startAngle: .pi / 2, endAngle: .pi, clockwise: false)
         }
 
         if case .left = pointer.edge {
-            addTriangle(triangle)
+            path.addTriangle(triangle)
         }
 
-        close()
+        path.closeSubpath()
+        return path
     }
+}
+
+private extension CGMutablePath {
+    func addTriangle(_ triangle: Triangle) {
+        if triangle.cornerRadius == 0 {
+            addLine(to: triangle.point1)
+            addLine(to: triangle.point2)
+            addLine(to: triangle.point3)
+        } else {
+            if triangle.offCenterPointer1 {
+                // Can't round the straight edge
+                addLine(to: triangle.point1)
+            } else {
+                addArc(tangent1End: triangle.point1, tangent2End: triangle.point2, radius: triangle.cornerRadius)
+            }
+
+            addArc(tangent1End: triangle.point2, tangent2End: triangle.point3, radius: triangle.cornerRadius)
 
-    private func addTriangle(_ triangle: Triangle) {
-        addLine(to: triangle.point1)
-        addLine(to: triangle.point2)
-        addLine(to: triangle.point3)
+            if triangle.offCenterPointer2 {
+                // Can't round the straight edge
+                addLine(to: triangle.point3)
+            } else {
+                addArc(tangent1End: triangle.point3, tangent2End: triangle.point4, radius: triangle.cornerRadius)
+            }
+        }
     }
-}
 
-private extension UIBezierPath {
     struct Triangle {
         // Points are ordered for a tooltip drawn clockwise
         let point1: CGPoint
         let point2: CGPoint
         let point3: CGPoint
 
+        let cornerRadius: CGFloat
+
+        // Control points for rounded pointer arc calculations
+        let point0: CGPoint
+        let point4: CGPoint
+
         private(set) var overridesTopLeftCorner: Bool
         private(set) var overridesTopRightCorner: Bool
         private(set) var overridesBottomRightCorner: Bool
         private(set) var overridesBottomLeftCorner: Bool
 
+        private(set) var offCenterPointer1: Bool
+        private(set) var offCenterPointer2: Bool
```

Comment:
curious, what do these represent exactly?

Reply:
Updated the naming to hopefully be clearer and also took the opportunity to rename the `Triangle` struct to be `Pointer.PathInfo` for clarity.

Reply:
thanks - updated names help I think. Adding that explanation above in the comments maybe helpful for future ref as well.